lib/dynflow/execution_plan/steps/abstract.rb

```diff
@@ -112,8 +112,8 @@ def progress_weight
 
       # @return [Action] in presentation mode, intended for retrieving: progress information,
       # details, human outputs, etc.
-      def action(execution_plan)
-        world.persistence.load_action_for_presentation(execution_plan, action_id)
+      def action(execution_plan, with_step = false)
+        world.persistence.load_action_for_presentation(execution_plan, action_id, with_step ? self : nil)
```

Comment:
Since it's the first case where the step data is needed, it seemed less intrusive to me to provide the step data on-demand instead of always, but I don't have strong opinions about this.